File: dualForceBackend/src/main/java/com/dualforce/dualForceBackend/entity/ExpertSystem.java
package com.dualforce.dualForceBackend.entity;

import com.dualforce.dualForceBackend.entity.documentEntities.DocumentMetadata;
import com.fasterxml.jackson.annotation.JsonBackReference;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import org.bson.types.ObjectId;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.index.Indexed;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.LocalDateTime;

@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Document(collection = "expert_system")
public class ExpertSystem {
    @Id
    private ObjectId id;

    private String name;

    private String description;

    private LocalDateTime createdAt;

    private LocalDateTime updatedAt;

    @DBRef
    private DocumentMetadata documentMetadata;

    @DBRef
    @JsonBackReference // Prevents circular reference during serialization
    private User user;
}

File: dual_force_frontend/lib/models/expert_system.dart
import 'package:dual_force/models/document_metadata.dart';

class ExpertSystem {
  final String id;
  final String name;
  final String description;
  final DateTime createdAt;
  final DateTime updatedAt;
  final DocumentMetadata documentMetadata;

  ExpertSystem({
    required this.id,
    required this.name,
    required this.description,
    required this.createdAt,
    required this.updatedAt,
    required this.documentMetadata,
  });

  Map<String, dynamic> toMap() {
    return {
      'id': id,
      'name': name,
      'description': description,
      'createdAt': createdAt,
      'updatedAt': updatedAt,
      'documentMetadata': documentMetadata.toMap(),
    };
  }

  factory ExpertSystem.fromMap(Map<String, dynamic> map) {
    return ExpertSystem(
      id: map['id'],
      name: map['name'],
      description: map['description'],
      createdAt: DateTime.parse(map['createdAt']),
      updatedAt: DateTime.parse(map['updatedAt']),
      documentMetadata: DocumentMetadata.fromMap(map['documentMetadata']),
    );
  }
}